src/cmd/cli/command/compose.go

```diff
@@ -501,99 +503,122 @@ func makeComposePsCmd() *cobra.Command {
 }
 
 func makeComposeLogsCmd() *cobra.Command {
-	logType := logs.LogTypeRun
 	var logsCmd = &cobra.Command{
 		Use:         "logs [SERVICE...]",
 		Annotations: authNeededAnnotation,
-		Aliases:     []string{"tail"},
 		Short:       "Show logs from one or more services",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			var name, _ = cmd.Flags().GetString("name")
-			var etag, _ = cmd.Flags().GetString("etag")
-			var deployment, _ = cmd.Flags().GetString("deployment")
-			var raw, _ = cmd.Flags().GetBool("raw")
-			var since, _ = cmd.Flags().GetString("since")
-			var utc, _ = cmd.Flags().GetBool("utc")
-			var verbose, _ = cmd.Flags().GetBool("verbose")
-			var filter, _ = cmd.Flags().GetString("filter")
-			var until, _ = cmd.Flags().GetString("until")
+		RunE:        handleLogsCmd,
+	}
+	setupLogsFlags(logsCmd)
+	return logsCmd
+}
 
-			if etag != "" && deployment == "" {
-				deployment = etag
-			}
+func makeComposeTailCmd() *cobra.Command {
+	var tailCmd = &cobra.Command{
+		Use:         "tail [SERVICE...]",
+		Annotations: authNeededAnnotation,
+		Short:       "Show logs from one or more services",
+		RunE:        handleLogsCmd,
+	}
+	setupLogsFlags(tailCmd)
+	return tailCmd
+}
 
-			if utc {
-				cli.EnableUTCMode()
-			}
+func setupLogsFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP("name", "n", "", "name of the service (backwards compat)")
+	cmd.Flags().MarkHidden("name")
+	cmd.Flags().String("etag", "", "deployment ID (ETag) of the service")
+	cmd.Flags().MarkHidden("etag")
+	cmd.Flags().String("deployment", "", "deployment ID of the service")
+	cmd.Flags().Bool("follow", false, "follow log output, --until will be ignored") // NOTE: -f is already used by --file
+	cmd.Flags().BoolP("raw", "r", false, "show raw (unparsed) logs")
+	cmd.Flags().String("since", "", "show logs since duration/time")
+	cmd.Flags().String("until", "", "show logs until duration/time")
+	cmd.Flags().Bool("utc", false, "show logs in UTC timezone (ie. TZ=UTC)")
+	cmd.Flags().Var(&logType, "type", fmt.Sprintf("show logs of type; one of %v", logs.AllLogTypes))
+	cmd.Flags().String("filter", "", "only show logs containing given text; case-insensitive")
+}
 
-			if !cmd.Flags().Changed("verbose") {
-				verbose = true // default verbose for explicit tail command
-			}
+func handleLogsCmd(cmd *cobra.Command, args []string) error {
+	var name, _ = cmd.Flags().GetString("name")
+	var etag, _ = cmd.Flags().GetString("etag")
+	var deployment, _ = cmd.Flags().GetString("deployment")
+	var raw, _ = cmd.Flags().GetBool("raw")
+	var since, _ = cmd.Flags().GetString("since")
+	var utc, _ = cmd.Flags().GetBool("utc")
+	var verbose, _ = cmd.Flags().GetBool("verbose")
+	var filter, _ = cmd.Flags().GetString("filter")
+	var until, _ = cmd.Flags().GetString("until")
+	var follow, _ = cmd.Flags().GetBool("follow")
+
+	if cmd.Name() == "tail" {
```

Comment:
instead of this "hack", with this new setup you can pass this "default" to the new `setupLogsFlags` function `defFollow bool`.